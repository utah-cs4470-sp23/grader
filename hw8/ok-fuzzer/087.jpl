let a[b] = args
let {} = (if true then (if (((! false) == (! false)) && {false}{0}) then (if ((! true) != false) then {} else (if (argnum > b) then {} else (if false then {} else {}))) else {}) else (if true then (if (b == b) then (array[c : argnum, d : 157, e : b] {}) else (array[c : 811, d : 823, e : b] {}))[b, (891 + (- b)), 169] else (array[c : (b + argnum), d : (- b), e : (b / 451)] {})[(- (if true then b else b)), (sum[c : (- b)] args[c]), ((b * b) - b)]))
fn c() : int {
  return argnum
  let {{d}} = {{(if (! true) then {args} else {a})}, {argnum}}{0}
  let e = log(sin((sin(acos(25.0)) - log(atan2(37.23, 10.11)))))
  return (- (array[f : b, g : b, h : argnum] {g, c(), {}})[(sum[f : c(), g : (sum[f : b] 821), h : (c() * argnum)] (- b)), {b}{0}, c()]{0})
  return c()
}

let {} = {{args, (! true), (false || true)}{1}, atan2(to_float(a[717]), (if (argnum >= b) then to_float(argnum) else to_float(b))), (if true then {argnum, {}}{1} else {})}{2}
let {d, e, f[g]} = {exp(pow(sqrt(acos(81.24)), cos(tan(35.65)))), (- pow(sin(exp(76.38)), {80.63}{0})), a}
assert (if (((c() > (- c())) != ((! true) || false)) || (! false)) then ({d, (c() < c()), e}{1} == (atan2(acos(e), to_float(c())) <= exp(exp(e)))) else (true && true)), "h"
show g

let {{h[i, j, k]}, l[m, n]} = (if ((asin(sqrt(91.93)) / sqrt(atan2(d, e))) > {(sum[h : g, i : 125, j : b] e), a, 73.48}{0}) then {{(array[h : c(), i : c(), j : argnum] 404)}, {c(), (array[h : g, i : g] args), false}{1}} else {{(array[h : (if true then argnum else g), i : b, j : (if false then b else g)] argnum)}, (array[h : to_int((e * e)), i : g] a)})
let {{}, o} = (if (if (! true) then (d < to_float(m)) else ((false || false) != true)) then {{{}, f}} else {{(if true then {} else {}), f}}){0}
let p = log((sum[p : (sum[p : c(), q : m] c()), q : argnum] d))
let q = c()
read image "r.png" to r

