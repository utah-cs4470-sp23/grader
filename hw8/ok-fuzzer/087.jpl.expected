(LetCmd (ArgLValue (ArrayArgument a b)) (VarExpr (ArrayType (IntType) 1) args))
(LetCmd
 (TupleLValue)
 (IfExpr
  (TupleType)
  (TrueExpr (BoolType))
  (IfExpr
   (TupleType)
   (BinopExpr
    (BoolType)
    (BinopExpr
     (BoolType)
     (UnopExpr (BoolType) ! (FalseExpr (BoolType)))
     ==
     (UnopExpr (BoolType) ! (FalseExpr (BoolType))))
    &&
    (TupleIndexExpr
     (BoolType)
     (TupleLiteralExpr (TupleType (BoolType)) (FalseExpr (BoolType)))
     0))
   (IfExpr
    (TupleType)
    (BinopExpr
     (BoolType)
     (UnopExpr (BoolType) ! (TrueExpr (BoolType)))
     !=
     (FalseExpr (BoolType)))
    (TupleLiteralExpr (TupleType))
    (IfExpr
     (TupleType)
     (BinopExpr (BoolType) (VarExpr (IntType) argnum) > (VarExpr (IntType) b))
     (TupleLiteralExpr (TupleType))
     (IfExpr
      (TupleType)
      (FalseExpr (BoolType))
      (TupleLiteralExpr (TupleType))
      (TupleLiteralExpr (TupleType)))))
   (TupleLiteralExpr (TupleType)))
  (IfExpr
   (TupleType)
   (TrueExpr (BoolType))
   (ArrayIndexExpr
    (TupleType)
    (IfExpr
     (ArrayType (TupleType) 3)
     (BinopExpr (BoolType) (VarExpr (IntType) b) == (VarExpr (IntType) b))
     (ArrayLoopExpr
      (ArrayType (TupleType) 3)
      c
      (VarExpr (IntType) argnum)
      d
      (IntExpr (IntType) 157)
      e
      (VarExpr (IntType) b)
      (TupleLiteralExpr (TupleType)))
     (ArrayLoopExpr
      (ArrayType (TupleType) 3)
      c
      (IntExpr (IntType) 811)
      d
      (IntExpr (IntType) 823)
      e
      (VarExpr (IntType) b)
      (TupleLiteralExpr (TupleType))))
    (VarExpr (IntType) b)
    (BinopExpr
     (IntType)
     (IntExpr (IntType) 891)
     +
     (UnopExpr (IntType) - (VarExpr (IntType) b)))
    (IntExpr (IntType) 169))
   (ArrayIndexExpr
    (TupleType)
    (ArrayLoopExpr
     (ArrayType (TupleType) 3)
     c
     (BinopExpr (IntType) (VarExpr (IntType) b) + (VarExpr (IntType) argnum))
     d
     (UnopExpr (IntType) - (VarExpr (IntType) b))
     e
     (BinopExpr (IntType) (VarExpr (IntType) b) / (IntExpr (IntType) 451))
     (TupleLiteralExpr (TupleType)))
    (UnopExpr
     (IntType)
     -
     (IfExpr
      (IntType)
      (TrueExpr (BoolType))
      (VarExpr (IntType) b)
      (VarExpr (IntType) b)))
    (SumLoopExpr
     (IntType)
     c
     (UnopExpr (IntType) - (VarExpr (IntType) b))
     (ArrayIndexExpr
      (IntType)
      (VarExpr (ArrayType (IntType) 1) args)
      (VarExpr (IntType) c)))
    (BinopExpr
     (IntType)
     (BinopExpr (IntType) (VarExpr (IntType) b) * (VarExpr (IntType) b))
     -
     (VarExpr (IntType) b))))))
(FnCmd
 c
 ()
 (IntType)
 (ReturnStmt (VarExpr (IntType) argnum))
 (LetStmt
  (TupleLValue (TupleLValue (ArgLValue (VarArgument d))))
  (TupleIndexExpr
   (TupleType (TupleType (ArrayType (IntType) 1)))
   (TupleLiteralExpr
    (TupleType
     (TupleType (TupleType (ArrayType (IntType) 1)))
     (TupleType (IntType)))
    (TupleLiteralExpr
     (TupleType (TupleType (ArrayType (IntType) 1)))
     (IfExpr
      (TupleType (ArrayType (IntType) 1))
      (UnopExpr (BoolType) ! (TrueExpr (BoolType)))
      (TupleLiteralExpr
       (TupleType (ArrayType (IntType) 1))
       (VarExpr (ArrayType (IntType) 1) args))
      (TupleLiteralExpr
       (TupleType (ArrayType (IntType) 1))
       (VarExpr (ArrayType (IntType) 1) a))))
    (TupleLiteralExpr (TupleType (IntType)) (VarExpr (IntType) argnum)))
   0))
 (LetStmt
  (ArgLValue (VarArgument e))
  (CallExpr
   (FloatType)
   log
   (CallExpr
    (FloatType)
    sin
    (BinopExpr
     (FloatType)
     (CallExpr
      (FloatType)
      sin
      (CallExpr (FloatType) acos (FloatExpr (FloatType) 25)))
     -
     (CallExpr
      (FloatType)
      log
      (CallExpr
       (FloatType)
       atan2
       (FloatExpr (FloatType) 37)
       (FloatExpr (FloatType) 10)))))))
 (ReturnStmt
  (UnopExpr
   (IntType)
   -
   (TupleIndexExpr
    (IntType)
    (ArrayIndexExpr
     (TupleType (IntType) (IntType) (TupleType))
     (ArrayLoopExpr
      (ArrayType (TupleType (IntType) (IntType) (TupleType)) 3)
      f
      (VarExpr (IntType) b)
      g
      (VarExpr (IntType) b)
      h
      (VarExpr (IntType) argnum)
      (TupleLiteralExpr
       (TupleType (IntType) (IntType) (TupleType))
       (VarExpr (IntType) g)
       (CallExpr (IntType) c)
       (TupleLiteralExpr (TupleType))))
     (SumLoopExpr
      (IntType)
      f
      (CallExpr (IntType) c)
      g
      (SumLoopExpr (IntType) f (VarExpr (IntType) b) (IntExpr (IntType) 821))
      h
      (BinopExpr (IntType) (CallExpr (IntType) c) * (VarExpr (IntType) argnum))
      (UnopExpr (IntType) - (VarExpr (IntType) b)))
     (TupleIndexExpr
      (IntType)
      (TupleLiteralExpr (TupleType (IntType)) (VarExpr (IntType) b))
      0)
     (CallExpr (IntType) c))
    0)))
 (ReturnStmt (CallExpr (IntType) c)))
(LetCmd
 (TupleLValue)
 (TupleIndexExpr
  (TupleType)
  (TupleLiteralExpr
   (TupleType (BoolType) (FloatType) (TupleType))
   (TupleIndexExpr
    (BoolType)
    (TupleLiteralExpr
     (TupleType (ArrayType (IntType) 1) (BoolType) (BoolType))
     (VarExpr (ArrayType (IntType) 1) args)
     (UnopExpr (BoolType) ! (TrueExpr (BoolType)))
     (BinopExpr (BoolType) (FalseExpr (BoolType)) || (TrueExpr (BoolType))))
    1)
   (CallExpr
    (FloatType)
    atan2
    (CallExpr
     (FloatType)
     to_float
     (ArrayIndexExpr
      (IntType)
      (VarExpr (ArrayType (IntType) 1) a)
      (IntExpr (IntType) 717)))
    (IfExpr
     (FloatType)
     (BinopExpr (BoolType) (VarExpr (IntType) argnum) >= (VarExpr (IntType) b))
     (CallExpr (FloatType) to_float (VarExpr (IntType) argnum))
     (CallExpr (FloatType) to_float (VarExpr (IntType) b))))
   (IfExpr
    (TupleType)
    (TrueExpr (BoolType))
    (TupleIndexExpr
     (TupleType)
     (TupleLiteralExpr
      (TupleType (IntType) (TupleType))
      (VarExpr (IntType) argnum)
      (TupleLiteralExpr (TupleType)))
     1)
    (TupleLiteralExpr (TupleType))))
  2))
(LetCmd
 (TupleLValue
  (ArgLValue (VarArgument d))
  (ArgLValue (VarArgument e))
  (ArgLValue (ArrayArgument f g)))
 (TupleLiteralExpr
  (TupleType (FloatType) (FloatType) (ArrayType (IntType) 1))
  (CallExpr
   (FloatType)
   exp
   (CallExpr
    (FloatType)
    pow
    (CallExpr
     (FloatType)
     sqrt
     (CallExpr (FloatType) acos (FloatExpr (FloatType) 81)))
    (CallExpr
     (FloatType)
     cos
     (CallExpr (FloatType) tan (FloatExpr (FloatType) 35)))))
  (UnopExpr
   (FloatType)
   -
   (CallExpr
    (FloatType)
    pow
    (CallExpr
     (FloatType)
     sin
     (CallExpr (FloatType) exp (FloatExpr (FloatType) 76)))
    (TupleIndexExpr
     (FloatType)
     (TupleLiteralExpr (TupleType (FloatType)) (FloatExpr (FloatType) 80))
     0)))
  (VarExpr (ArrayType (IntType) 1) a)))
(AssertCmd
 (IfExpr
  (BoolType)
  (BinopExpr
   (BoolType)
   (BinopExpr
    (BoolType)
    (BinopExpr
     (BoolType)
     (CallExpr (IntType) c)
     >
     (UnopExpr (IntType) - (CallExpr (IntType) c)))
    !=
    (BinopExpr
     (BoolType)
     (UnopExpr (BoolType) ! (TrueExpr (BoolType)))
     ||
     (FalseExpr (BoolType))))
   ||
   (UnopExpr (BoolType) ! (FalseExpr (BoolType))))
  (BinopExpr
   (BoolType)
   (TupleIndexExpr
    (BoolType)
    (TupleLiteralExpr
     (TupleType (FloatType) (BoolType) (FloatType))
     (VarExpr (FloatType) d)
     (BinopExpr (BoolType) (CallExpr (IntType) c) < (CallExpr (IntType) c))
     (VarExpr (FloatType) e))
    1)
   ==
   (BinopExpr
    (BoolType)
    (CallExpr
     (FloatType)
     atan2
     (CallExpr (FloatType) acos (VarExpr (FloatType) e))
     (CallExpr (FloatType) to_float (CallExpr (IntType) c)))
    <=
    (CallExpr
     (FloatType)
     exp
     (CallExpr (FloatType) exp (VarExpr (FloatType) e)))))
  (BinopExpr (BoolType) (TrueExpr (BoolType)) && (TrueExpr (BoolType))))
 "h")
(ShowCmd (VarExpr (IntType) g))
(LetCmd
 (TupleLValue
  (TupleLValue (ArgLValue (ArrayArgument h i j k)))
  (ArgLValue (ArrayArgument l m n)))
 (IfExpr
  (TupleType
   (TupleType (ArrayType (IntType) 3))
   (ArrayType (ArrayType (IntType) 1) 2))
  (BinopExpr
   (BoolType)
   (BinopExpr
    (FloatType)
    (CallExpr
     (FloatType)
     asin
     (CallExpr (FloatType) sqrt (FloatExpr (FloatType) 91)))
    /
    (CallExpr
     (FloatType)
     sqrt
     (CallExpr
      (FloatType)
      atan2
      (VarExpr (FloatType) d)
      (VarExpr (FloatType) e))))
   >
   (TupleIndexExpr
    (FloatType)
    (TupleLiteralExpr
     (TupleType (FloatType) (ArrayType (IntType) 1) (FloatType))
     (SumLoopExpr
      (FloatType)
      h
      (VarExpr (IntType) g)
      i
      (IntExpr (IntType) 125)
      j
      (VarExpr (IntType) b)
      (VarExpr (FloatType) e))
     (VarExpr (ArrayType (IntType) 1) a)
     (FloatExpr (FloatType) 73))
    0))
  (TupleLiteralExpr
   (TupleType
    (TupleType (ArrayType (IntType) 3))
    (ArrayType (ArrayType (IntType) 1) 2))
   (TupleLiteralExpr
    (TupleType (ArrayType (IntType) 3))
    (ArrayLoopExpr
     (ArrayType (IntType) 3)
     h
     (CallExpr (IntType) c)
     i
     (CallExpr (IntType) c)
     j
     (VarExpr (IntType) argnum)
     (IntExpr (IntType) 404)))
   (TupleIndexExpr
    (ArrayType (ArrayType (IntType) 1) 2)
    (TupleLiteralExpr
     (TupleType (IntType) (ArrayType (ArrayType (IntType) 1) 2) (BoolType))
     (CallExpr (IntType) c)
     (ArrayLoopExpr
      (ArrayType (ArrayType (IntType) 1) 2)
      h
      (VarExpr (IntType) g)
      i
      (VarExpr (IntType) g)
      (VarExpr (ArrayType (IntType) 1) args))
     (FalseExpr (BoolType)))
    1))
  (TupleLiteralExpr
   (TupleType
    (TupleType (ArrayType (IntType) 3))
    (ArrayType (ArrayType (IntType) 1) 2))
   (TupleLiteralExpr
    (TupleType (ArrayType (IntType) 3))
    (ArrayLoopExpr
     (ArrayType (IntType) 3)
     h
     (IfExpr
      (IntType)
      (TrueExpr (BoolType))
      (VarExpr (IntType) argnum)
      (VarExpr (IntType) g))
     i
     (VarExpr (IntType) b)
     j
     (IfExpr
      (IntType)
      (FalseExpr (BoolType))
      (VarExpr (IntType) b)
      (VarExpr (IntType) g))
     (VarExpr (IntType) argnum)))
   (ArrayLoopExpr
    (ArrayType (ArrayType (IntType) 1) 2)
    h
    (CallExpr
     (IntType)
     to_int
     (BinopExpr (FloatType) (VarExpr (FloatType) e) * (VarExpr (FloatType) e)))
    i
    (VarExpr (IntType) g)
    (VarExpr (ArrayType (IntType) 1) a)))))
(LetCmd
 (TupleLValue (TupleLValue) (ArgLValue (VarArgument o)))
 (TupleIndexExpr
  (TupleType (TupleType) (ArrayType (IntType) 1))
  (IfExpr
   (TupleType (TupleType (TupleType) (ArrayType (IntType) 1)))
   (IfExpr
    (BoolType)
    (UnopExpr (BoolType) ! (TrueExpr (BoolType)))
    (BinopExpr
     (BoolType)
     (VarExpr (FloatType) d)
     <
     (CallExpr (FloatType) to_float (VarExpr (IntType) m)))
    (BinopExpr
     (BoolType)
     (BinopExpr (BoolType) (FalseExpr (BoolType)) || (FalseExpr (BoolType)))
     !=
     (TrueExpr (BoolType))))
   (TupleLiteralExpr
    (TupleType (TupleType (TupleType) (ArrayType (IntType) 1)))
    (TupleLiteralExpr
     (TupleType (TupleType) (ArrayType (IntType) 1))
     (TupleLiteralExpr (TupleType))
     (VarExpr (ArrayType (IntType) 1) f)))
   (TupleLiteralExpr
    (TupleType (TupleType (TupleType) (ArrayType (IntType) 1)))
    (TupleLiteralExpr
     (TupleType (TupleType) (ArrayType (IntType) 1))
     (IfExpr
      (TupleType)
      (TrueExpr (BoolType))
      (TupleLiteralExpr (TupleType))
      (TupleLiteralExpr (TupleType)))
     (VarExpr (ArrayType (IntType) 1) f))))
  0))
(LetCmd
 (ArgLValue (VarArgument p))
 (CallExpr
  (FloatType)
  log
  (SumLoopExpr
   (FloatType)
   p
   (SumLoopExpr
    (IntType)
    p
    (CallExpr (IntType) c)
    q
    (VarExpr (IntType) m)
    (CallExpr (IntType) c))
   q
   (VarExpr (IntType) argnum)
   (VarExpr (FloatType) d))))
(LetCmd (ArgLValue (VarArgument q)) (CallExpr (IntType) c))
(ReadCmd "r.png" (VarArgument r))
